test/MatrixFields/scalar_fieldmatrix.jl

```diff
@@ -0,0 +1,273 @@
+    end
+    struct two_fields{T1, T2}
+        x::T1
+        y::T2
+    end
+    function test_broadcasted_get_field_type(
+        name,
+        ::Type{T},
+        expected_type,
+    ) where {T}
+        @test_all MatrixFields.broadcasted_get_field_type(T, name) ==
+                  expected_type
+    end
+    test_broadcasted_get_field_type(
+        @name(x),
+        singleton{singleton{singleton{singleton{Float64}}}},
+        singleton{singleton{singleton{Float64}}},
+    )
+    test_broadcasted_get_field_type(
+        @name(x.x.x),
+        singleton{singleton{singleton{singleton{Float64}}}},
+        singleton{Float64},
+    )
+    test_broadcasted_get_field_type(
+        @name(y.x),
+        two_fields{
+            two_fields{Float64, Float64},
+            two_fields{Float64, two_fields{Float64, singleton{Float64}}},
+        },
+        Float64,
+    )
+    test_broadcasted_get_field_type(
+        @name(y.y.y),
+        two_fields{
+            two_fields{Float64, Float64},
+            two_fields{Float64, two_fields{Float64, singleton{Float64}}},
+        },
+        singleton{Float64},
+    )
+end
+
+@testset "fieldmatrix to scalar fieldmatrix unit tests" begin
+    FT = Float64
+    center_space, face_space = test_spaces(FT)
+    surface_space = Spaces.level(face_space, half)
+    seed!(1)
+    ᶜvec = random_field(FT, center_space)
+    ᶠvec = random_field(FT, face_space)
+    sfc_vec = random_field(FT, surface_space)
+    λ = 10
+    ᶜᶜmat1 = random_field(DiagonalMatrixRow{FT}, center_space) ./ λ .+ (I,)
+    ᶜᶠmat2 = random_field(BidiagonalMatrixRow{FT}, center_space) ./ λ
+    ᶠᶜmat2 = random_field(BidiagonalMatrixRow{FT}, face_space) ./ λ
+    ᶜᶜmat3 = random_field(TridiagonalMatrixRow{FT}, center_space) ./ λ .+ (I,)
+    ᶠᶠmat3 = random_field(TridiagonalMatrixRow{FT}, face_space) ./ λ .+ (I,)
+
+    e¹² = Geometry.Covariant12Vector(1, 2)
+    e³ = Geometry.Covariant3Vector(1)
+    e₃ = Geometry.Contravariant3Vector(1)
+
+    ρχ_unit = (; ρq_tot = 1, ρq_liq = 1, ρq_ice = 1, ρq_rai = 1, ρq_sno = 1)
+    ρaχ_unit =
+        (; ρaq_tot = 1, ρaq_liq = 1, ρaq_ice = 1, ρaq_rai = 1, ρaq_sno = 1)
+
+    dry_center_gs_unit = (; ρ = 1, ρe_tot = 1, uₕ = e¹²)
+    center_gs_unit = (; dry_center_gs_unit..., ρatke = 1, ρχ = ρχ_unit)
+    center_sgsʲ_unit = (; ρa = 1, ρae_tot = 1, ρaχ = ρaχ_unit)
+
+    ᶜᶜmat3_uₕ_scalar = ᶜᶜmat3 .* (e¹²,)
+    ᶠᶜmat2_u₃_scalar = ᶠᶜmat2 .* (e³,)
+    ᶜᶠmat2_scalar_u₃ = ᶜᶠmat2 .* (e₃',)
+    ᶜᶜmat3_scalar_uₕ = ᶜᶜmat3 .* (e¹²,)
+    ᶜᶜmat1_scalar_uₕ = ᶜᶜmat1 .* (e¹²,)
+    ᶜᶠmat2_uₕ_u₃ = ᶜᶠmat2 .* (e¹² * e₃',)
+    ᶠᶠmat3_u₃_u₃ = ᶠᶠmat3 .* (e³ * e₃',)
+    ᶜᶜmat3_ρχ_scalar = map(Base.Fix1(map, Base.Fix2(⊠, ρχ_unit)), ᶜᶜmat3)
+    ᶜᶜmat3_ρaχ_scalar = map(Base.Fix1(map, Base.Fix2(⊠, ρaχ_unit)), ᶜᶜmat3)
+    ᶜᶠmat2_ρχ_u₃ = map(Base.Fix1(map, Base.Fix2(⊠, ρχ_unit ⊠ e₃')), ᶜᶠmat2)
+    ᶜᶠmat2_ρaχ_u₃ = map(Base.Fix1(map, Base.Fix2(⊠, ρaχ_unit ⊠ e₃')), ᶜᶠmat2)
+
+    A = MatrixFields.FieldMatrix(
```

Comment:
You might want to move the definition of this `FieldMatrix` into `matrix_fields_test_utils.jl` so that we don't have so much duplicate code